### PR TITLE
Add index to `history_date` field to improve performance of `get_next_record()` and `get_prev_record()` methods

### PR DESCRIPTION
**Pull request by nex2hex**
Based on https://github.com/jazzband/django-simple-history/pull/601

## Description
Calling methods `get_next_record()` and `get_prev_record()`  https://github.com/treyhunner/django-simple-history/blob/master/simple_history/models.py#L391 creates a heavy load on database without indexes (full scan)

## Related Issue
https://github.com/treyhunner/django-simple-history/issues/565

## How Has This Been Tested?
Create new migration after changes and test sql queries time

## Types of changes
- [ x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ x] I have run the `make format` command to format my code
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] I have added my name and/or github handle to `AUTHORS.rst`
- [x ] I have added my change to `CHANGES.rst`
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by nex2hex** [comment]
updated to master from https://github.com/jazzband/django-simple-history/pull/601

**Comment by nex2hex** [comment]
@rossmechanic I'm done

File: simple_history/models.py [hunk 1]
```diff
@@ -421,7 +421,7 @@ def get_default_history_user(instance):
 
         extra_fields = {
             "history_id": self._get_history_id_field(),
-            "history_date": models.DateTimeField(),
+            "history_date": models.DateTimeField(db_index=True),
```
Comment (rossmechanic) [inline simple_history/models.py:424]: Can we make this a setting that is False by default in django settings? And then can we add a param that allows for model level override?

Comment (nex2hex) [reply simple_history/models.py:424]: ok

**Pull request closed without merging**